ignite/contrib/metrics/gpu_info.py

```diff
@@ -1,100 +1,91 @@
-# -*- coding: utf-8 -*-
-import warnings
+from threading import Thread
+from time import sleep, time
+from pynvml import nvmlInit, nvmlDeviceGetCount, nvmlDeviceGetHandleByIndex, nvmlDeviceGetMemoryInfo, \
+    nvmlDeviceGetName, nvmlDeviceGetUtilizationRates
+from torch.utils.tensorboard import SummaryWriter
 
-import torch
 
-from ignite.metrics import Metric
-from ignite.engine import Events
-
-
-class GpuInfo(Metric):
-    """Provides GPU information: a) used memory percentage, b) gpu utilization percentage values as Metric
-    on each iterations.
-
-    Examples:
-
-        .. code-block:: python
-
-            # Default GPU measurements
-            GpuInfo().attach(trainer, name='gpu')  # metric names are 'gpu:X mem(%)', 'gpu:X util(%)'
-
-            # Logging with TQDM
-            ProgressBar(persist=True).attach(trainer, metric_names=['gpu:0 mem(%)', 'gpu:0 util(%)'])
-            # Progress bar will looks like
-            # Epoch [2/10]: [12/24]  50%|█████      , gpu:0 mem(%)=79, gpu:0 util(%)=59 [00:17<1:23]
-
-            # Logging with Tensorboard
-            tb_logger.attach(trainer,
-                             log_handler=OutputHandler(tag="training", metric_names='all'),
-                             event_name=Events.ITERATION_COMPLETED)
+class GpuPynvmlLogger(Thread):
```

Comment:
Already discovered the first bugs... the pytest wasn't running for other reasons:
`GpuPynvmlLogger` -> `GpuInfo`